packages/flutter/lib/src/services/text_input.dart

```diff
@@ -721,10 +721,34 @@ TextAffinity? _toTextAffinity(String? affinity) {
   return null;
 }
 
-/// A floating cursor state the user has induced by force pressing an iOS
-/// keyboard.
+/// The state of a "floating cursor" drag on an iOS soft keyboard.
+///
+/// The "floating cursor" cursor-positioning mode is an iOS feature used to
+/// precisely position the caret in some editable text. When the user
+/// long-presses the spacebar on the iOS virtual keyboard, iOS enters floating
+/// cursor mode where the whole keyboard becomes a trackpad. In this mode,
+/// there are two visible cursors. One, the floating cursor, hovers over the
+/// text, following the user's horizontal movements exactly and snapping to
```

Comment:
The "floating" cursor follows the user's gesture movements exactly I believe (there's no line-snapping).

Reply:
I tried it again and it does sort of snap. Maybe that's not the best way to call it?

![output](https://github.com/flutter/flutter/assets/389558/9b187a3e-be76-41c1-b176-43563311d3b5)

Reply:
Is this in a UIKit app? I don't think Flutter does this and I don't remember UIKit doing this either. It could be new.

Reply:
Also both cursors being visible when they are so close to each other is different from the behavior I remembered. Pretty sure it used to hide the gray cursor when it's too close to the blue cursor.

Reply:
This is the Reminders app on an iOS 17 simulator. I see the same behavior on Messages so maybe it's a new iOS 17 thing.